lib/katello/tasks/upgrades/3.0/update_subscription_facet_backend_data.rake

```diff
@@ -1,10 +1,40 @@
 namespace :katello do
   namespace :upgrades do
     namespace '3.0' do
-      task :update_subscription_facet_backend_data => ["environment"] do
+      task :update_subscription_facet_backend_data => ["environment"]  do
+
+        def error(message)
+          @errors << message
+          @errors << "\n"
+        end
+
+        def report_errors
+          if @errors.any?
+            file = File.open("/tmp/upgrade-#{Time.now.to_i}", 'w')
```

Comment:
/var/log/foreman-installer/ isn't writable by the foreman user, but /var/log/foreman/ is.  However that isn't available in dev, so i'll add both (and check if /var/log/foreman/ is writable)